CRM/Contribute/Form/ContributionRecur.php

```diff
@@ -109,14 +116,20 @@ public function getDefaultContext() {
 
   /**
    * Set variables up before form is built.
+   *
+   * @throws \CRM_Core_Exception
    */
   public function preProcess() {
     $this->setAction(CRM_Core_Action::UPDATE);
     $this->_mid = CRM_Utils_Request::retrieve('mid', 'Integer', $this, FALSE);
     $this->_crid = CRM_Utils_Request::retrieve('crid', 'Integer', $this, FALSE);
     $this->contributionRecurID = $this->_crid;
     $this->_coid = CRM_Utils_Request::retrieve('coid', 'Integer', $this, FALSE);
+    $this->setSubscriptionDetails();
     $this->setPaymentProcessor();
+    if ($this->getSubscriptionDetails()['contact_id']) {
+      $this->set('cid', $this->getSubscriptionDetails()['contact_id']);
```

Comment:
is this an array or an object? looking at https://github.com/civicrm/civicrm-core/pull/14388/files#diff-30a8e5b11544ab10573dea3f36cb9db2R357 it suggests object? @eileenmcnaughton

Reply:
I think looking at https://github.com/civicrm/civicrm-core/blob/master/CRM/Contribute/BAO/ContributionRecur.php#L387 it should be 

```suggestion
      $this->set('cid', $this->getSubscriptionDetails->contact_id);
```